server/src/main/java/io/druid/segment/realtime/appenderator/AppenderatorImpl.java

```diff
@@ -307,20 +319,22 @@ public void clear() throws InterruptedException
     // Drop commit metadata, then abandon all segments.
 
     try {
-      final ListenableFuture<?> uncommitFuture = persistExecutor.submit(
-          new Callable<Object>()
-          {
-            @Override
-            public Object call() throws Exception
+      if (persistExecutor != null) {
```

Comment:
I saw a test failure due to NPE from here, but can't find what it was.. Maybe there were some bugs which are fixed now.